extensions/amp-analytics/0.1/linker-manager.js

```diff
@@ -311,6 +322,20 @@ export class LinkerManager {
     return true;
   }
 
+  /**
+   * Allows specified wildcard matching of domains.
+   * Example:
+   *    `*.foo.com` matches `amp.foo.com`
+   *    `*.foo.com*` matches `amp.foo.com.uk`
```

Comment:
We need to clarify the following cases, and add unit tests to cover them:
- does `*.foo.co*` match `me.foo.com`?
- does `*.foo.com` match `foo.com`?
- does `*.foo.com` match `a.b.foo.com`?

Needs GA's input on this, what do you expect? @zikas @avimehta

Reply:
As it is currently written:
> * does `*.foo.co*` match `me.foo.com`?
Yes
> * does `*.foo.com` match `foo.com`?
No (no dot)
> * does `*.foo.com` match `a.b.foo.com`?
Yes

Happy to change if not in line with expectations.

Reply:
That all sounds good to me. @avimehta @zikas to confirm.

Can we have a set of tests for this private method? Just to cover all the corner cases, and also express the expectations.

Reply:
lgtm. I think it is supported but just to confirm:

`*.foo.co*` should match `me.foo.co.uk` as well.

Reply:
@avimehta yes that will match, added test case.

Also added unit tests specifically targeting this logic.